src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -48,16 +49,120 @@ public async Task BestLocations_ReturnsOK(DateTimeOffset start, DateTimeOffset e
         _dataSourceMocker.SetupDataMock(start, end, location);
 
         //Call the private method to construct with parameters
-        var endpointURI = ConstructDateQueryURI(bestLocationsURI, location, start, end);
+        var queryStrings = new Dictionary<string, string>();
+        queryStrings["location"] = location;
+        queryStrings["time"] = $"{start:O}";
+        queryStrings["toTime"] = $"{end:O}";
+
+        var endpointURI = ConstructUriWithQueryString(bestLocationsURI,queryStrings);
 
         //Get response and response content
         var result = await _client.GetAsync(endpointURI);
 
         Assert.That(result, Is.Not.Null);
         Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+    }
+
+    [TestCase("location", "", TestName = "empty location query string")]
+    [TestCase("non-location-param", "", TestName = "location param not present")]
+    public async Task BestLocations_EmptyLocationQueryString_ReturnsBadRequest(string queryString, string value)
+    {
+        //Call the private method to construct with parameters
+        var queryStrings = new Dictionary<string, string>();
+        queryStrings[queryString] = value;
+
+        var endpointURI = ConstructUriWithQueryString(bestLocationsURI, queryStrings);
+
+        //Get response and response content
+        var result = await _client.GetAsync(endpointURI);
+
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+    }
+
+    [Test]
+    public async Task EmissionsForecastsCurrent_JSON_ReturnsNotImplemented()
+    {
+        var ignoredDataSources = new List<DataSourceType>() { DataSourceType.WattTime };
```

Comment:
You've done this a couple times, might be good to extract to a helper? You pass the datasource type and do the check in the same helper

Reply:
Something to add here: Not sure why there is an ignore list and if feels confusing. Maybe something explicit would be easier to follow.